### ISSUE TITLE
Install emacs D mode?

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#1361]
I recently noticed that `D-mode.el` exists (and apparently has for a very long time).  However, it's not installed and the corresponding lines in `M2-init.el` are commented out by default.

Is there a reason it's not installed?  There's the possibility of a name clash, e.g., with [d-mode](https://melpa.org/#/d-mode), especially on systems where filenames are case insensitive.  Perhaps a rename to something like `M2-D-mode.el`?

**Comment by DanGrayson** [comment]
Yes, I haven't used it in years.  (I just activate M2-mode on *.d files.)   Renaming it and providing it would be good.  Checking that it behaves well would also be good.  (Notice that the file is now in a submodule.)

**Comment by mahrud** [comment]
The distributed M2 packages don't carry .d/dd files or the scc1 compiler, so there's no need to install the emacs mode for those files.

Developers, of course, can still install it manually from the source repository.

**Comment by DanGrayson** [comment]
Good point.

**Comment by d-torrance** [comment]
Personally, I like `apt` to install everything for me so I can just get to coding without fiddling with installing more things manually.  :)

But if Dan uses M2 mode, does anyone use D mode?  Perhaps it should just disappear completely?  There's a lot of overlap anyway.  Maybe add some of the D-only keywords to M2-symbols.el?

**Comment by DanGrayson** [comment]
It would be nice if D-mode were made to work -- then I would use it again.

**Comment by DanGrayson** [comment]
One fix needed: 
```diff
--- a/D-mode.el
+++ b/D-mode.el
@@ -109,7 +109,7 @@
 
 (defun D-electric-semi ()
      (interactive)
-     (insert ?;)
+     (insert ?\;)
      (and (eolp) (next-line-blank) (= 0 (paren-change))
 	 (D-newline-and-indent))
      )
```
